morphia/src/test/java/org/mongodb/morphia/IndexHelperTest.java

```diff
@@ -0,0 +1,443 @@
+        Index index = new IndexBuilder()
+            .fields(new FieldBuilder()
+                        .value("indexName"),
+                    new FieldBuilder()
+                        .value("text")
+                        .type(IndexType.DESC))
+            .options(indexOptions());
+        indexHelper.createIndex(indexes, mappedClass, index, false);
+        List<DBObject> indexInfo = getDs().getCollection(IndexedClass.class)
+                                          .getIndexInfo();
+        for (DBObject dbObject : indexInfo) {
+            if (dbObject.get("name").equals("indexName")) {
+                checkIndex(dbObject);
+
+                assertEquals("en", dbObject.get("default_language"));
+                assertEquals("de", dbObject.get("language_override"));
+
+                assertEquals(new BasicDBObject()
+                                 .append("locale", "en")
+                                 .append("caseLevel", true)
+                                 .append("caseFirst", "upper")
+                                 .append("strength", 5)
+                                 .append("numericOrdering", true)
+                                 .append("alternate", "shifted")
+                                 .append("maxVariable", "space")
+                                 .append("backwards", true)
+                                 .append("normalization", true)
+                                 .append("version", "57.1"),
+                             dbObject.get("collation"));
+            }
+        }
+    }
+
+    @Test
+    public void indexCollationConversion() {
+        Collation collation = collation();
+        com.mongodb.client.model.Collation driver = indexHelper.convert(collation);
+        assertEquals("en", driver.getLocale());
+        assertTrue(driver.getCaseLevel());
+        assertEquals(UPPER, driver.getCaseFirst());
+        assertEquals(IDENTICAL, driver.getStrength());
+        assertTrue(driver.getNumericOrdering());
+        assertEquals(SHIFTED, driver.getAlternate());
+        assertEquals(SPACE, driver.getMaxVariable());
+        assertTrue(driver.getNormalization());
+        assertTrue(driver.getBackwards());
+    }
+
+    @Test
+    public void indexOptionsConversion() {
+        IndexOptionsBuilder indexOptions = indexOptions();
+        com.mongodb.client.model.IndexOptions options = indexHelper.convert(indexOptions, false);
+        assertEquals("index_name", options.getName());
+        assertTrue(options.isBackground());
+        assertTrue(options.isUnique());
+        assertTrue(options.isSparse());
+        assertEquals(Long.valueOf(42), options.getExpireAfter(TimeUnit.SECONDS));
+        assertEquals("en", options.getDefaultLanguage());
+        assertEquals("de", options.getLanguageOverride());
+        assertEquals(indexHelper.convert(indexOptions.collation()), options.getCollation());
+
+        assertTrue(indexHelper.convert(indexOptions, true).isBackground());
```

Comment:
it's just testing that values copy over correctly.  My big complaint is that we'll have to manually track evolutions of the object manually.  I'm pondering some sort of reflection based test so that methods are discovered and tested automatically.  That way we don't have to remember to update the tests when new values show up.  But that's for another time.